platform_google.py

```diff
@@ -309,12 +422,140 @@ def get_subscription_v2(package_name: str, token: str, err: base.ErrorSink) -> S
 def handle_not_implemented(name: str, err: base.ErrorSink):
     err.msg_list.append(f"'{name}' is not implemented!")
 
+def get_line_item(details: SubscriptionV2Data, err: base.ErrorSink):
+    result = None
+    if len(details.line_items) == 0:
+        err.msg_list.append(f"No line items for subscription!")
+    else:
+        line_item = details.line_items[0]
+        if line_item is None:
+            err.msg_list.append(f"No line item for subscription!")
+        else:
+            result = line_item
+    return result
+
+def queue_user_entitlement_amend(purchase_token: str, details: SubscriptionV2Data, err: base.ErrorSink):
+    line_item = get_line_item(details, err)
+
+    if err.has():
+        err.msg_list.append(f'Failed to add user entitlement for {purchase_token}')
+        return
+
+    details = get_subscription_details_for_plan(line_item.product_id, err)
+
+    if err.has():
+        err.msg_list.append(f'Failed to add user entitlement for {purchase_token}')
+        return
+
+    tx = PaymentProviderTransaction(
+        provider=base.PaymentProvider.GooglePlayStore,
+        google_order_id=line_item.offer_details.offer_id,
+        google_payment_token=purchase_token,
+    )
+
+    sql_conn = sqlite3.connect(env.SESH_PRO_BACKEND_DB_PATH)
```

Comment:
This is leaking connections I'm pretty sure, needs to be scoped. I can't remember if connect implements \_\_exit__, if not you can use OpenDBAtPath to make this exception safe.